doc/sphinx/language/core/variants.rst

```diff
@@ -6,11 +6,29 @@ Variants and the `match` construct
 Variants
 --------
 
-.. cmd:: Variant @ident_decl {* @binder } {? %| {* @binder } } {? : @type } := {? %| } {+| @constructor } {? @decl_notations }
-
-   The :cmd:`Variant` command is similar to the :cmd:`Inductive` command, except
-   that it disallows recursive definition of types (for instance, lists cannot
-   be defined using :cmd:`Variant`). No induction scheme is generated for
+The :cmd:`Variant` command allows defining types by listing
+the :term:`inhabitants` of the type. This includes enumerated types from
+programming languages, such as Booleans, characters or even the
+degenerate cases of the unit and empty types. Variant types more
+generally include enumerated types with arguments or even enumerated
+types with parametric arguments such as option types, sum types,
+etc. It also includes predicates or type families defined by cases
+such as the Boolean reflection or equality predicates. Observing the
+form of the term:`inhabitants` of a variant type is done by case analysis
+using the :cmd:`match` expression. Each case is described by what is
+called a :gdef:`constructor` of the type. For instance, Booleans have two
+constructors: :g:`true` and :g:`false`.
+
+When a constructor of a type takes an argument of that same type,
+the type becomes recursive, in which case it can be either
+:cmd:`Inductive` or :cmd:`CoInductive`. The keyword :cmd:`Variant`
+is reserved for non-recursive types. Natural numbers, lists or streams cannot
+be defined using :cmd:`Variant`.
+
+.. cmd:: Variant @ident {? @cumul_univ_decl } {* @binder } {? %| {* @binder } } {? : @type } := {? %| } {+| @constructor } {? @decl_notations }
```

Comment:
There are 15 uses of `ident_decl` in the refman. If you think that it creates confusion and should be unfolded, we should do it for all the occurrences (at the risk of making grammar description longer). You can test this change by adding `ident_decl` to the SPLICE list of `common.edit_mlg` (currently starting on line 2260) and running `make doc_gram_rsts`. This change which doesn't match what is in `common.edit_mlg` / `orderedGrammar` is the reason for the linter failure.

Reply:
I was needing a way to refer to the name "ident" of the variant type. I followed the example of the Inductive section where ident_decl was already split. I have to particular preference otherwise.

Reply:
It is split in `Inductive` because it uses `cumul_univ_decl` which is not used anywhere else. Typically the way we refer to production names that do not appear directly is to write "the `ident` (in `ident_decl`)".

Reply:
I see. Done. Hopefully as expected. If not, I can still change.